run.sh

```diff
@@ -36,383 +38,280 @@ core config:
-  if [ "$MULTISITE" == "true" ]; then
-    h3 "Multisite found. Enabling..."
-    WP core multisite-convert |& loglevel
-    STATUS "${PIPESTATUS[0]}"
-  else
-    h3 "Multisite not found. SKIPPING..."
-    STATUS SKIP
-  fi
+# FIXME: Remove in next version
+# Deprecations
+# ---------------------
+_local_deprecation() {
+  local local_type="$1" # 'plugin' or 'theme'
+  echo "Warning: [local]$local_type-name has been deprecated and will be dropped in the next version." |& _colorize
+}
 
-  h2 "Checking themes"
-  check_themes
+_search_replace_depreaction() {
+  echo "Warning: SEARCH_REPLACE environment variable has been renamed to URL_REPLACE and will be dropped in the next version." |& _colorize
+}
 
-  h2 "Checking plugins"
-  check_plugins
+# Config Functions
+# ---------------------
 
-  h2 "Finalizing"
-  if [ ! -f /app/.htaccess ]; then
-    h3 "Generating .htaccess file"
-    if [[ "$MULTISITE" == 'true' ]]; then
-      STATUS 1
-      h3warn "Cannot generate .htaccess for multisite!"
-    else
-      WP rewrite structure "$PERMALINKS" |& loglevel
-      WP rewrite flush --hard |& loglevel
-      STATUS "${PIPESTATUS[0]}"
-    fi
+init() {
+  local plugins themes i IFS=$'\n'
+
+  # FIXME: Remove in next version
+  [[ -n $SEARCH_REPLACE ]] && _search_replace_depreaction
+
+  # Download WordPress
+  # ------------------
+  if [[ ! -f /app/wp-settings.php ]]; then
+    h2 "Downloading WordPress"
+    _wp core download --version="$WP_VERSION"
+  fi
+
+  PLUGINS="${PLUGINS/%,},"
+  THEMES="${THEMES/%,},"
+
+  if [[ -f /app/.dockercache ]]; then
+    . /app/.dockercache
   else
-    h3 ".htaccess exists. SKIPPING..."
-    STATUS SKIP
+    plugins=$( _get_volumes plugins )
+    themes=$( _get_volumes themes )
+    echo "plugins='$plugins'" >> /app/.dockercache
+    echo "themes='$themes'" >> /app/.dockercache
   fi
 
-  h3 "Adjusting file permissions"
-  groupadd -f docker && usermod -aG docker www-data
-  find /app -type d -exec chmod 755 {} \;
-  find /app -type f -exec chmod 644 {} \;
-  mkdir -p /app/wp-content/uploads
-  chmod -R 775 /app/wp-content/uploads && \
-    chown -R :docker /app/wp-content/uploads
-  STATUS $?
+  while read -r i; do
+    [[ ! "$i" ]] && continue
+    plugin_volumes[$i]="$i"
+  done <<< "$plugins"
+
+  while read -r i; do
+    [[ ! "$i" ]] && continue
+    theme_volumes[$i]="$i"
+  done <<< "$themes"
+
+  local key value
+
+  while read -r -d, i; do
+    [[ ! "$i" ]] && continue
+    i="${i# }"          # Trim leading whitespace
+    key="${i%]*}"       # Trim right bracket to end of string
+    key="${key//[\[ ]}" # Trim left bracket
+    value="${i##\[*\]}" # Trim bracketed text inclusive
+    # FIXME: Remove in next version
+    [[ "$key" == 'local' ]] && _local_deprecation plugin && continue
+    plugin_deps[$key]="$value"
+  done <<< "$PLUGINS"
 
-  h1 "WordPress Configuration Complete!"
+  while read -r -d, i; do
+    [[ ! "$i" ]] && continue
+    i="${i# }"          # Trim leading whitespace
```

Comment:
Actually, I take that back. It would end up being more code than just keeping as is because bash functions can't return values. So there'd have to be a minimum of 2 side effects each time it's called, which I'm not super fond of. 

I can still change it if you're firm on this, but I wouldn't recommend it.